sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/SentEventNotifier.java

```diff
@@ -71,24 +76,18 @@ public void notify(EventObject event) {
             return;
         }
 
-        if (currentSpan.isRemote()) {
-            log.debug("Span {} is remote log :ss: event.", currentSpan);
-            currentSpan.logEvent(Span.SERVER_SEND);
-            //need to be unset as remote in order to be sent for collection
-            currentSpan = currentSpan.toBuilder().remote(false).build();
-        } else {
-            log.debug("Span {} is not remote log :cr: event.", currentSpan);
-            currentSpan.logEvent(Span.CLIENT_RECV);
-        }
-
-        logExceptionIfExists(event, currentSpan);
+        exchange.removeProperty(EXCHANGE_IS_TRACED_BY_BRAVE);
 
-        tracer.close(currentSpan);
-        log.debug("Span {} successfully closed.", currentSpan);
+        Span spanToFinish = threadLocalSpan.remove();
+        logExceptionIfExists(event, spanToFinish);
+        spanToFinish.annotate(EXCHANGE_EVENT_SENT_ANNOTATION);
+        spanToFinish.finish();
+        log.debug("Span {} successfully closed.", spanToFinish);
     }
 
-    private boolean isCamelSpan(Span span) {
-        return span.tags().containsKey(FROM_CAMEL);
+    private boolean isCamelSpan(Exchange exchange) {
```

Comment:
Why not just `return exchange.getProperty(EXCHANGE_IS_TRACED_BY_BRAVE, Boolean.class) `

Reply:
it can be null